src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,242 @@
+const DigitTicker = (() => {
+    let el_container,
+        el_peek,
+        el_peek_box,
+        el_mask,
+        total_tick_count,
+        contract_status,
+        current_spot;
+    let digit_block_size = 36;
+    let style_offset_correction = 5;
+
+    const array_of_digits         = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+    const init = (container_id, contract_type, barrier, tick_count, status = 'open') => {
+        contract_status      = status;
+        total_tick_count     = tick_count;
+        current_spot         = '-';
+        el_container         = document.querySelector(`#${container_id}`);
+        populateContainer(el_container);
+        highlightWinningNumbers(getWinningNumbers(contract_type, barrier));
+        observeResize();
+    };
+
+    const populateContainer = (container_element) => {
+        // remove previous elements and start fresh.
+        while (container_element.firstChild) {
+            container_element.removeChild(container_element.firstChild);
+        }
+
+        const temp_epoch_el = document.createElement('div');
```

Comment:
Just a nitpick perhaps, but would it be better to have the `DOM` elements created in a static template for this?

Reply:
Since we render these dynamically, the option of having them embedded in the `view-popup` or `smart trader` will make additional DOM elements that are not required by all purchase types. In `app_2`, I think creating the template is a sane choice, tho 👍